jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/PartitionAlgorithm.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.interfaces;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Algorithm to compute a vertex partition of a graph.
+ *
+ * @param <V> vertex the graph vertex type
+ *
+ * @author Alexandru Valeanu
+ * @since June 2018
+ */
+public interface PartitionAlgorithm<V> {
+
+    /**
+     * Computes a vertex partition.
+     *
+     * @return a vertex partition
+     */
+    Partition<V> getPartition();
+
+    /**
+     * Check if the given vertex partition is valid.
+     *
+     * @param partition the input vertex partition
+     * @return true if the input partition is valid, false otherwise
+     */
+    boolean isValidPartition(Partition<V> partition);
+
+    /**
+     * A <a href="https://en.wikipedia.org/wiki/Graph_partition">graph partition</a>.
+     *
+     * @param <V> the vertex type
+     */
+    interface Partition<V>{
```

Comment:
it would be nice to have Paritition<V> extend 
        Iterable<Set<V>> so it's possible to immediately iterate over the partitions.